src/Domain/Domain.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Domain.
+
+#pragma once
+
+#include <iosfwd>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/DomainHelpers.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+template <typename SourceFrame, typename TargetFrame, size_t Dim>
+class CoordinateMapBase;
+
+/*!
+ *  \ingroup ComputationalDomain
+ *  A wrapper around a vector of Blocks that represent the computational domain.
+ */
+template <size_t VolumeDim, typename TargetFrame>
+class Domain {
+ public:
+  explicit Domain(std::vector<Block<VolumeDim, TargetFrame>>&& blocks);
+
+  /*!
+   * Create a Domain using a corner numbering scheme to encode the Orienations,
+   * with an optional parameter that encodes periodic boundary conditions.
+   */
+  Domain(std::vector<std::unique_ptr<
+             CoordinateMapBase<Frame::Logical, TargetFrame, VolumeDim>>>&& maps,
+         const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+             corners_of_all_blocks,
+         const std::vector<std::vector<size_t>>& identifications = {});
+
+  ~Domain() = default;
+  Domain(const Domain&) = delete;
+  Domain(Domain&&) = delete;
+  Domain<VolumeDim, TargetFrame>& operator=(
+      const Domain<VolumeDim, TargetFrame>&) = delete;
+  Domain<VolumeDim, TargetFrame>& operator=(Domain<VolumeDim, TargetFrame>&&) =
+      delete;
+
+  const std::vector<Block<VolumeDim, TargetFrame>>& blocks() const {
+    return blocks_;
+  }
+
+ private:
```

Comment:
Presumably we need a pup?

Reply:
Likely we will, and the pup function is trivial since both std::vector and Block are pupable.